lib/widget/button.dart

```diff
@@ -89,25 +92,37 @@ class ButtonState extends WidgetState<Button> {
     bool isOutlineButton = widget._controller.outline ?? false;
 
     ScopeManager? scopeManager = DataScopeWidget.getScope(context);
-    Widget label;
-    if (scopeManager != null && widget._controller.labelWidget != null) {
-      label = scopeManager.buildWidgetWithScopeFromDefinition(widget._controller.labelWidget);
-    } else {
-      label = Text(
-          Utils.translate(widget._controller.label ?? '', context));
+    Widget? startingIcon;
+    Widget? endingIcon;
+    if (widget._controller.startingIcon != null) {
+      startingIcon = scopeManager?.buildWidgetFromDefinition(
+          widget._controller.startingIcon);
+    }
+    if (widget._controller.endingIcon != null) {
+      endingIcon = scopeManager?.buildWidgetFromDefinition(widget._controller.endingIcon);
+    }
+    List<Widget> labelParts = [Text(Utils.translate(widget._controller.label ?? '', context))];
```

Comment:
User may want to control the gap between the icons and label. Add an `iconGap` attribute? by default it should be 0.

Reply:
This is easy enough to add, but I think this is a slippery slope where we might consider going back to a custom label widget Row so they can control things like spacing without needing to pixel push. Do you feel strongly we should add this now or can we defer this until we get some more user feedback?